esp/tools/soapplus/httpproxy.cpp

```diff
@@ -674,25 +672,28 @@ class CSocksProxyThread : public Thread
                 return -1;
             }
 
-            len += lenread;
-            m_client->read(inbuf + len, 0, 1, lenread);
+            char inbuf2[16];
+            m_client->read(inbuf2, 0, 16, lenread);
             StringBuffer username;
             while(lenread > 0)
             {
-                len += lenread;
-                if(len >= 1023)
+                bool done = false;
+                for (int i = lenread - 1; i >= 0; i--)
                 {
-                    len = 0;
+                    if (inbuf2[i] == '\0')
+                    {
+                        if (i > 0)
+                            username.append(i, inbuf2);
+                        done = true;
+                        break;
+                    }
                 }
-                if(inbuf[len - 1] == '\0')
-                {
+                if (done)
                     break;
```

Comment:
Why check for done and break here ?

Reply:
Seeing 0 means got all data expected.

Reply:
But you checked for that already in the if block above

Reply:
That's for the for loop, this check is to exit the outer while loop...

Reply:
ok, yes now I understand - sorry I missed that its needed to exit the while